library/src/main/java/ch/decent/sdk/api/MessagingApi.kt

```diff
@@ -0,0 +1,45 @@
+package ch.decent.sdk.api
+
+import ch.decent.sdk.DCoreApi
+import ch.decent.sdk.crypto.Credentials
+import ch.decent.sdk.model.*
+import ch.decent.sdk.net.model.request.GetMessageObjects
+import io.reactivex.Single
+import io.reactivex.functions.BiFunction
+
+class MessagingApi internal constructor(api: DCoreApi) : BaseApi(api) {
+
+  fun getAll(sender: ChainObject? = null, receiver: ChainObject? = null, maxCount: Int = 1000) =
+      GetMessageObjects(sender, receiver, maxCount).toRequest()
+
+  fun createMessageOperation(
```

Comment:
Not a review comment, but a point for discussion.

I think these methods should actually be `internal` and they should not be available for third party developers. Instead they should have a method such as `sendMessage` or something similar. I believe all Operation objects should be `internal`.

Reply:
Yes the plan is to have methods like sendMessage (as transfer already is) but operations should also be available, there might be cases when client wants to send multiple operations in single transaction.

Reply:
...since transaction costs DCT, one could bulk multiple operations into transaction